openstack/compute/v2/extensions/secgroups/requests.go

```diff
@@ -108,7 +108,7 @@ func DeleteWithRetry(c *golangsdk.ServiceClient, id string, timeout int) error {
 	return golangsdk.WaitFor(timeout, func() (bool, error) {
 		_, err := c.Delete(resourceURL(c, id), nil)
 		if err != nil {
-			if _, ok := err.(golangsdk.ErrDefault409); ok {
+			if _, ok := err.(golangsdk.ErrDefault400); ok {
```

Comment:
This is not a part of SFS, please move to another PR. Also, maybe check for both 409 and 400?